src/Components/PhoneNumberPaymentInput.res

```diff
@@ -1,37 +1,88 @@
-open RecoilAtoms
-open PaymentType
-
 @react.component
 let make = () => {
+  open RecoilAtoms
+  open PaymentType
+  open Utils
+
+  let phoneRef = React.useRef(Nullable.null)
   let {fields} = Recoil.useRecoilValueFromAtom(optionAtom)
   let loggerState = Recoil.useRecoilValueFromAtom(loggerAtom)
-
-  let (phone, setPhone) = Recoil.useLoggedRecoilState(userPhoneNumber, "phone", loggerState)
-
   let showDetails = getShowDetails(~billingDetails=fields.billingDetails, ~logger=loggerState)
-  let formatBSB = bsb => {
-    let formatted = bsb
+  let (phone, setPhone) = Recoil.useLoggedRecoilState(userPhoneNumber, "phone", loggerState)
+  let clientCountry = getClientCountry(CardUtils.dateTimeFormat().resolvedOptions().timeZone)
+  let currentCountryCode = Utils.getCountryCode(clientCountry.countryName)
+  let (displayValue, setDisplayValue) = React.useState(_ => "")
 
-    let secondPart = formatted->String.sliceToEnd(~start=4)->String.trim
+  let countryAndCodeCodeList =
+    phoneNumberJson
+    ->JSON.Decode.object
+    ->Option.getOr(Dict.make())
+    ->getArray("countries")
 
-    if formatted->String.length <= 4 {
-      "+351 "
-    } else if formatted->String.length > 4 {
-      `+351 ${secondPart}`
-    } else {
-      formatted
+  let phoneNumberCodeOptions: array<
+    DropdownField.optionType,
+  > = countryAndCodeCodeList->Array.reduce([], (acc, countryObj) => {
+    let countryObjDict = countryObj->getDictFromJson
+    let phoneNumberOptionsValue: DropdownField.optionType = {
+      label: `${countryObjDict->getString("country_flag", "")} ${countryObjDict->getString(
```

Comment:
Please write these `countryObjDict->getString("country_flag", "")` separately in a variable and then use inside label and displayValue

Reply:
Updated